// Description:
// This patch (which shows up in three files listed below) allows dialogs generated
// by the Dialog Editor (.wxr files) to be used in wxWizard pages. Currently
// if multiple dialog pages are generated and LoadFromResources() is used the library
// attempts to load all the controls on top of one another resulting in an 
// unusable mess.
// This patch was originally generated using the 2.2.6 stable release and is
// being used in an application built with a modified 2.2.6 wxMSW.
// Robert Cavanaugh
//   Flextronics Semiconductor
//   Salt Lake City
//   bob.cavanaugh@flextronics.com

--- wxWindows/include/wx/wizard.h	Sat Sep 15 04:39:10 2001
+++ wxWindowsnew/include/wx/wizard.h	Fri Oct 12 13:15:50 2001
@@ -5,5 +5,6 @@
 // Author:      Vadim Zeitlin (partly based on work by Ron Kuris and Kevin B.
 //              Smith)
-// Modified by:
+// Modified by:	Robert Cavanaugh
+//              Added capability to use .WXR resource files in Wizard pages
 // Created:     15.08.99
 // RCS-ID:      $Id: wizard.h,v 1.13 2001/09/15 10:39:10 VS Exp $
@@ -49,5 +50,9 @@
     // that no other parameters are needed because the wizard will resize and
     // reposition the page anyhow
-    wxWizardPage(wxWizard *parent, const wxBitmap& bitmap = wxNullBitmap);
+    // Modified 8/20/2001 Robert Cavanaugh
+	// Add an optional resource string to build the page controls
+    wxWizardPage(wxWizard *parent, 
+                const wxBitmap& bitmap = wxNullBitmap,
+                char* resource = NULL);
 
     // these functions are used by the wizard to show another page when the
--- wxWindows/src/generic/wizard.cpp	Fri Aug 17 11:52:02 2001
+++ wxWindowsnew/src/generic/wizard.cpp	Thu Oct 11 15:47:14 2001
@@ -3,5 +3,6 @@
 // Purpose:     generic implementation of wxWizard class
 // Author:      Vadim Zeitlin
-// Modified by:
+// Modified by: Robert Cavanaugh
+//              Added capability for wxWizardPage to accept resources
 // Created:     15.08.99
 // RCS-ID:      $Id: wizard.cpp,v 1.19 2001/08/17 17:52:02 VZ Exp $
@@ -74,7 +75,18 @@
 // ----------------------------------------------------------------------------
 
-wxWizardPage::wxWizardPage(wxWizard *parent, const wxBitmap& bitmap)
-            : wxPanel(parent), m_bitmap(bitmap)
+wxWizardPage::wxWizardPage(wxWizard *parent, 
+                            const wxBitmap& bitmap,
+                            char *resource)
+            : wxPanel(parent),m_bitmap(bitmap)
 {
+    // Added 8/20/2001 Robert Cavanaugh
+    // Allow loading controls from dialog editor scripts
+    if(resource != NULL)
+    {
+        if((LoadFromResource(this, resource)) == FALSE)
+        {
+            wxFAIL_MSG(wxT("wxWizardPage LoadFromResource failed!!!!"));
+        }
+    }
     // initially the page is hidden, it's shown only when it becomes current
     Hide();
--- wxWindows/src/common/resource.cpp	Sat Aug 18 11:17:14 2001
+++ wxWindowsnew/src/common/resource.cpp	Fri Oct 12 13:25:42 2001
@@ -3,5 +3,6 @@
 // Purpose:     Resource system
 // Author:      Julian Smart
-// Modified by:
+// Modified by: Robert Cavanaugh
+//              Allow LoadFromResources() to be used with Dialog Editor scripts
 // Created:     04/01/98
 // RCS-ID:      $Id: resource.cpp,v 1.77 2001/08/18 17:17:13 RR Exp $
@@ -3146,4 +3147,9 @@
     wxString name = resource->GetName();
 
+    // Added 8/21/2001 Robert Cavanaugh
+    // allows windows to accept resources in the
+    // class constructor
+    if(this != parent)
+    {
     if (IsKindOf(CLASSINFO(wxDialog)))
     {
@@ -3167,4 +3173,5 @@
         if (!((wxWindow *)this)->Create(parent, -1, wxPoint(x, y), wxSize(width, height), theWindowStyle, name))
             return FALSE;
+        }
     }


 	  	 
